File: Fortran/ex4/predict.f90
	subroutine predict(M, N, M1, N1, M2, N2, X, Theta1, Theta2, g)

	implicit none 
	
	integer::M, N, M1, N1, M2, N2
	double precision,dimension(M,N)::X
	double precision,dimension(M,N1)::X1
	double precision,dimension(M,N2)::gg
	
	double precision,dimension(M)::g
	double precision::sig
	double precision,dimension(M1,N1)::Theta1
	double precision,dimension(M2,N2)::Theta2
	double precision::TX, max, sigmoid
	integer::i, k, j
	
	
	do i=1,M
	   X1(i,1)=1
	   do j=2, N1
	   	X1(i,j) = X(i,j-1)
	   enddo
	enddo
	

	
	do i=1, M
	   do k=1, M1
	   	TX = 0;
		do j=1, N1
			TX = TX + X1(i,j)*Theta1(k,j)
		enddo
		
		gg(i,k+1) = sigmoid(TX)
	    enddo
	      gg(i,1) = 1
	 enddo
		 
!	write(*,*) ' ok here 2'
	
	
	do i=1, M
	       max = -1
		 
	      do k=1, M2
	          TX = 0
		    do j=1, N2
			TX = TX + gg(i,j)* Theta2(k,j)
		    enddo
		    
		    sig = sigmoid(TX)
		    
		    if(sig.gt.max)then
		       max = sig
			 g(i) = k
			 if(g(i).eq.0)g(i)=10.
		   endif
		    
		enddo
								 
	enddo

		
	 end subroutine predict

File: C/ex3/predict.c
#include <stdio.h>
double sigmoid(double z);

int predict(int M, int N, int M1, int N1, int M2, int N2, double X[M][N], double Theta1[M1][N1],double Theta2[M2][N2], double g[M])
{

        double TX, ggg, max;
        int i, j, k;
	  static double gg[5000][26];
	  static double X1[5000][401];
  
	  
	  for(i=0; i < M; i++){
	       X1[i][0] = 1;
		 for(j=1; j < N1; j++)
		    X1[i][j] = X[i][j-1];
	   }
		 

        for(i=0; i < M; i++){	      
	       for(k=0; k < M1; k++){
	           TX = 0;
                 for(j=0; j < N1; j++)
                    TX += X1[i][j] * Theta1[k][j];
		     
                 gg[i][k+1] = sigmoid(TX);
		 }		 		
	           gg[i][0] = 1;
        }
		
		
		
	     for(i=0; i < M; i++){
	        max = -1;
		  
	       for(k=0; k < M2; k++){
	           TX = 0;
                 for(j=0; j < N2; j++)
                      TX += gg[i][j] * Theta2[k][j];
		     
                 ggg = sigmoid(TX);		     	   
		   
		     if(ggg >  max ){
			  max = ggg;
		        g[i] = k+1;
			  if(g[i] == 0)g[i]=10;      
		     }
	        }	     	     
           }	  

        return 0 ;
}
